src/BlockTypes/FeaturedItem.php

```diff
@@ -0,0 +1,339 @@
+		if ( ! $item ) {
+			return '';
+		}
+
+		$attributes = wp_parse_args( $attributes, $this->defaults );
+
+		$attributes['height'] = $attributes['height'] ?? wc_get_theme_support( 'featured_block::default_height', 500 );
+
+		$image_url = esc_url( $this->get_image_url( $attributes, $item ) );
+
+		$styles  = $this->get_styles( $attributes );
+		$classes = $this->get_classes( $attributes );
+
+		$output  = sprintf( '<div class="%1$s wp-block-woocommerce-%2$s" style="%3$s">', esc_attr( trim( $classes ) ), $this->block_name, esc_attr( $styles ) );
+		$output .= sprintf( '<div class="wc-block-%s__wrapper">', $this->block_name );
+		$output .= $this->render_overlay( $attributes );
+
+		if ( ! $attributes['isRepeated'] && ! $attributes['hasParallax'] ) {
+			$output .= $this->render_image( $attributes, $item, $image_url );
+		} else {
+			$output .= $this->render_bg_image( $attributes, $image_url );
+		}
+
+		$output .= $this->render_attributes( $item, $attributes );
+		$output .= sprintf( '<div class="wc-block-%s__link">%s</div>', $this->block_name, $content );
+		$output .= '</div>';
+		$output .= '</div>';
+
+		return $output;
+	}
+
+	/**
+	 * Returns the url the item's image
+	 *
+	 * @param array                $attributes Block attributes. Default empty array.
+	 * @param \WP_Term|\WC_Product $item Item object.
+	 *
+	 * @return string
+	 */
+	private function get_image_url( $attributes, $item ) {
+		$image_size = 'large';
+		if ( 'none' !== $attributes['align'] || $attributes['height'] > 800 ) {
+			$image_size = 'full';
+		}
+
+		if ( $attributes['mediaId'] ) {
+			return wp_get_attachment_image_url( $attributes['mediaId'], $image_size );
+		}
+
+		return $this->get_item_image( $item, $image_size );
+	}
+
+	/**
+	 * Renders the featured image as a div background.
+	 *
+	 * @param array  $attributes Block attributes. Default empty array.
+	 * @param string $image_url Item image url.
+	 *
+	 * @return string
+	 */
+	private function render_bg_image( $attributes, $image_url ) {
+		$styles = $this->get_bg_styles( $attributes, $image_url );
+
+		$classes = [ "wc-block-{$this->block_name}__background-image" ];
+
+		if ( $attributes['hasParallax'] ) {
+			$classes[] = ' has-parallax';
+		}
+
+		return sprintf( '<div class="%1$s" style="%2$s" /></div>', implode( ' ', $classes ), $styles );
+	}
+
+	/**
+	 * Get the styles for the wrapper element (background image, color).
+	 *
+	 * @param array  $attributes Block attributes. Default empty array.
+	 * @param string $image_url Item image url.
+	 *
+	 * @return string
+	 */
+	public function get_bg_styles( $attributes, $image_url ) {
+		$style = '';
+
+		if ( $attributes['isRepeated'] || $attributes['hasParallax'] ) {
+			$style .= "background-image: url($image_url);";
+		}
+
+		if ( ! $attributes['isRepeated'] ) {
+			$style .= 'background-repeat: no-repeat;';
+			$style .= 'background-size: ' . ( 'cover' === $attributes['imageFit'] ? $attributes['imageFit'] : 'auto' ) . ';';
```

Comment:
What do you think of extracting this ternary operator outside of the string concatenation for a bit more readability? It's not awfully unreadable, so I'd leave it to your judgement if you find it better like this.